src/bind/solution/solution.cpp

```diff
@@ -49,32 +52,41 @@ void init_solution(py::module_ &m){
 
           ptr[idx].vehicle_id = route.vehicle;
 
-          if (step.step_type == vroom::STEP_TYPE::START) task = "start";
-          else if (step.step_type == vroom::STEP_TYPE::END) task = "end";
-          else if (step.step_type == vroom::STEP_TYPE::BREAK) task = "break";
-          else if (step.job_type == vroom::JOB_TYPE::SINGLE) task = "single";
-          else if (step.job_type == vroom::JOB_TYPE::PICKUP) task = "pickup";
-          else if (step.job_type == vroom::JOB_TYPE::DELIVERY) task = "delivery";
-          strncpy(ptr[idx].task, task.c_str(), 9);
+          if (step.step_type == vroom::STEP_TYPE::START) type = "start";
+          else if (step.step_type == vroom::STEP_TYPE::END) type = "end";
+          else if (step.step_type == vroom::STEP_TYPE::BREAK) type = "break";
+          else if (step.job_type == vroom::JOB_TYPE::SINGLE) type = "job";
+          else if (step.job_type == vroom::JOB_TYPE::PICKUP) type = "pickup";
+          else if (step.job_type == vroom::JOB_TYPE::DELIVERY) type = "delivery";
 
-          ptr[idx].job_id = step.id;
-          ptr[idx].arrival = step.arrival;
-          ptr[idx].duration = step.duration;
-          ptr[idx].waiting_time = step.waiting_time;
+          strncpy(ptr[idx].type, type.c_str(), 9);
+          strncpy(ptr[idx].description, step.description.c_str(), 40);
 
-          if (step.location.has_coordinates()) {
-            ptr[idx].loc_lon = step.location.lon();
-            ptr[idx].loc_lat = step.location.lat();
-          }
-          ptr[idx].loc_index = step.location.index();
+          ptr[idx].longitude = step.location.has_coordinates() ? step.location.lon() : NA_SUBSTITUTE;
+          ptr[idx].latitude = step.location.has_coordinates() ? step.location.lat() : NA_SUBSTITUTE;
+          ptr[idx].location_index = step.location.user_index() ? step.location.index() : NA_SUBSTITUTE;
 
-          strncpy(ptr[idx].description, step.description.c_str(), 40);
+          ptr[idx].id = (step.step_type == vroom::STEP_TYPE::JOB or
+                         step.step_type == vroom::STEP_TYPE::BREAK) ? step.id : NA_SUBSTITUTE;
+
+          ptr[idx].setup = step.setup;
+          ptr[idx].service = step.service;
+          ptr[idx].waiting_time = step.waiting_time;
+          ptr[idx].arrival = step.arrival;
+          ptr[idx].duration = step.duration;
 
           idx++;
         }
       }
       return arr;
     })
+    .def("_solution_json", [](vroom::Solution solution){
+        auto json_output = vroom::io::to_json(solution, false);
+        rapidjson::StringBuffer s;
+        rapidjson::Writer<rapidjson::StringBuffer> r_writer(s);
+        json_output.Accept(r_writer);
+        return s.GetString();
```

Comment:
@jcoupey, would you know why this snippet would return an empty string?

I copied the code from your `output_json.cpp`.

Reply:
No idea, that's weird as it looks exactly identical to the writing process from `io::write_to_json`.